0000-spec-distribution.md

```diff
@@ -0,0 +1,313 @@
+# Meta 
+[meta]: #meta
+- Name: Buildpack Distribution Specification
+- Start Date: 2019-04-12
+- CNB Pull Requests: (spec PR to follow)
+- CNB Issues: (lifecycle issues to follow)
+
+
+# Motivation
+[motivation]: #motivation
+
+This proposal enables both decentralized, manual distribution of buildpacks from Docker registries as well as automatic distribution of buildpacks from a centralized buildpack registry.
+It allows individual buildpacks to be distributed via a Docker registry, and it makes dynamic assembly of builders on a Docker registry efficient.
+It provides a developer-friendly interface that abstracts away complex buildpack configurations.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes an official way to distribute buildpacks that conform to the CNB buildpack v3 specification.
+Changes would consist of a new Buildpack Distribution specification, modifications to the lifecycle builder, and modifications to the pack CLI.
+
+It affects all personas that interact with buildpacks.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+## Overview of Changes
+
+- The `order.toml` and `buildpack.toml` files are merged into `buildpack.toml`.
+- `buildpack.toml` can contain multiple buildpacks.
+- A buildpack inside of `buildpack.toml` is either:
+  a. a single buildpack implementation specified by a `path` field or
+  b. an ordering of other buildpack IDs/versions
```

Comment:
Can we compose buildpackages from each other?

Reply:
It looks like yes (https://github.com/buildpack/rfcs/pull/11/files#diff-8e4e8df213b3904199aec818ffac0c54R249) unless I'm misunderstanding was "package"  here means.

Reply:
Yep, see the `[[packages]]` section in `package.toml`.

Reply:
So basically have a tree where nodes are either buildpacks or trees of buildpacks...interesting.

Reply:
👍